src/components/global/Header/Header.tsx

```diff
@@ -0,0 +1,66 @@
+import React, {
+  FunctionComponent,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import { BookSearch } from "../../BookFinder/BookSearch/BookSearch";
+import axios from "axios";
+import { searchUrl } from "../../../helpers/BookHelper";
+import { notification } from "antd";
+import { searchContext } from "../../../contexts/SearchContext/SearchContext";
+import { useHistory, useLocation } from "react-router-dom";
+import styles from "./Header.module.scss";
+
+export const Header: FunctionComponent = () => {
+  const [query, setQuery] = useState("");
+  const history = useHistory();
+  const location = useLocation();
+  const { search, setSearch } = useContext(searchContext);
+
+  const onSearch = (queryTerm: string) => {
+    const pathname = location.pathname;
+    if (pathname !== "/") {
+      history.push("/");
+    }
+    setSearch({ ...search, startIndex: 0 });
+    setQuery(queryTerm);
+  };
+
+  const fetchBooks = useCallback(async () => {
+    setSearch({ ...search, isLoading: true });
+
+    try {
+      const result = await axios(
+        `${searchUrl}${query}&startIndex=${search.startIndex}`
+      );
+
+      setSearch({
+        ...search,
+        books: result.data.items,
+        total: result.data.totalItems,
+        isLoading: false,
+      });
+    } catch (e) {
+      notification.error({
+        message: "An error occurred",
+        description: "Please try searching again",
+      });
+    }
+
+    window.scrollTo(0, 0);
+  }, [query, search.startIndex, setSearch]);
+
+  useEffect(() => {
+    if (query !== "") {
+      fetchBooks();
+    }
+  }, [fetchBooks, query]);
```

Comment:
Problem crasha ti je u ovom.

useEffect radi na svaku promjenu dependency arraya. kad prvi put promijenis query, useEffect se okine, ali se i promijeni fetchBooks. Posto fetchBooks promijeni search vrijednost, takoder invalidira samog sebe sto na kraju napravi ponovni poziv useEffecta!

Rješenje je eventualno drugačije modelirati context i razdvojiti loading state tako da ne invalidira tvoj fetchBooks ili koristiti useDeepMemo iz useDeepCompare liba. To ti je mozda trenutno najjednostavnije rješenje.

Dakle imao bi nešto poput :

``` 
const memoizedSearch = useDeepMemo(() => search, [search]);
```
dependency array za fetchBooks bi onda bio: ` [query, memoizedSerch, setSearch]`